src-db/database/sourcedata/AD_AUXILIARINPUT.xml

```diff
@@ -2417,6 +2417,17 @@ AND r.IsValidated = 'Y']]></CODE>
 <!--6351A2D1288F4F36A8B08F92374C7D2B-->  <AD_MODULE_ID><![CDATA[0]]></AD_MODULE_ID>
 <!--6351A2D1288F4F36A8B08F92374C7D2B--></AD_AUXILIARINPUT>
 
+<!--6351AB9CC115489C9147885C56CBCB8B--><AD_AUXILIARINPUT>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <AD_AUXILIARINPUT_ID><![CDATA[6351AB9CC115489C9147885C56CBCB8B]]></AD_AUXILIARINPUT_ID>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <AD_CLIENT_ID><![CDATA[0]]></AD_CLIENT_ID>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <AD_ORG_ID><![CDATA[0]]></AD_ORG_ID>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <ISACTIVE><![CDATA[Y]]></ISACTIVE>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <AD_TAB_ID><![CDATA[800049]]></AD_TAB_ID>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <NAME><![CDATA[OnHandLocatorDefault]]></NAME>
+<!--6351AB9CC115489C9147885C56CBCB8B-->  <CODE><![CDATA[@SQL=SELECT M_LOCATOR_ID FROM M_LOCATOR WHERE ISACTIVE='Y' AND M_WAREHOUSE_ID = (SELECT M_WAREHOUSE_ID FROM M_INOUT WHERE M_INOUT_ID = @M_INOUT_ID@) ORDER BY M_LOCATOR.ISDEFAULT DESC]]></CODE>
```

Comment:
You should consider moving from an IN to an EXISTS:

`SELECT ML.M_LOCATOR_ID FROM M_LOCATOR ML WHERE ML.ISACTIVE='Y' AND EXISTS (SELECT 1 FROM M_INOUT MI WHERE MI.M_INOUT_ID = @M_INOUT_ID@ AND MI.M_WAREHOUSE_ID = ML.M_WAREHOUSE_ID) ORDER BY ML.ISDEFAULT DESC;`

This is because:

1. Clarity of logic.
2. Potential efficiency by avoiding explicit retrieval of a value with SELECT.
3. Greater robustness against multiple result problems in the subquery.

Also, does it make sense to check for null? maybe adding a coalesce? think about whether it is possible and add it if necessary.

Reply:
I did it that way because this same query is done in two other windows and I copied it because that is what I needed to solve the bug. Anyway, now I will try the above mentioned.